app/assets/javascripts/recipes.js

```diff
@@ -11,15 +9,22 @@ $(document).ready(function(){
     });
     if(index > 0 && inputsAreBlank){
       $(grainElement).hide();
-    } else if (index > 0) {
-      $(grainElement).prev(".grain").find("button.add-grain").hide();
-      // Only show button for last non-blank input
     }
-   });
+  });
+
+  var show_last_add_grain_button = function(){
+    $('.grain').filter( function(){ 
+      if($(this).css('display') == "block") {
```

Comment:
The jQuery `:visible` selector extension may be useful here to save the `display == block` check, and make the code a little more understandable and concise. http://api.jquery.com/visible-selector/

Note that `:visible` selector is a jQuery extension and not available in CSS.

Also note that `:visible` can have a performance cost. The jQuery docs (linked above) recommend using it after do the initial element selection with a `filter(":visible")` call. So instead of: 
```javascript
$(".grain:visible")
```
prefer this:
```javascript
$(".grain").filter(":visible")
```

It's less clear to me if the `:last-child` selector might also be useful here: http://api.jquery.com/last-child-selector/